core/src/main/java/org/apache/carbondata/core/scan/collector/impl/RestructureBasedRawResultCollector.java

```diff
@@ -88,7 +88,7 @@ private void initRestructuredKeyGenerator() {
           // partitioner index will be 1 every column will be in columnar format
           updatedDimensionPartitioner.add(1);
           // for direct dictionary 4 bytes need to be allocated else 1
-          if (queryDimensions[i].getDimension().hasEncoding(Encoding.DIRECT_DICTIONARY)) {
+          if (queryDimensions[i].getDimension().getDataType() == DataTypes.DATE) {
```

Comment:
it will be alway true because it the  same with line 76, it should be optimized.

Reply:
fixed